simulators/ethereum/engine/suites/cancun/tests.go

```diff
@@ -537,6 +537,235 @@ var Tests = []test.SpecInterface{
+			- Payload Attributes uses Shanghai timestamp
+			- Payload Attributes' Beacon Root is nil
+
+			Verify that client returns INVALID_PARAMS_ERROR.
+			`,
+		},
+
+		CancunForkHeight: 2,
+
+		TestSequence: TestSequence{
+			NewPayloads{
+				FcUOnPayloadRequest: &helper.UpgradeForkchoiceUpdatedVersion{
+					ForkchoiceUpdatedCustomizer: &helper.BaseForkchoiceUpdatedCustomizer{
+						ExpectedError: INVALID_PARAMS_ERROR,
+					},
+				},
+				ExpectationDescription: fmt.Sprintf(`
+				ForkchoiceUpdatedV3 before Cancun with any nil field must return INVALID_PARAMS_ERROR (code %d)
+				`, *INVALID_PARAMS_ERROR),
+			},
+		},
+	},
+
+	&CancunBaseSpec{
+		Spec: test.Spec{
+			Name: "ForkchoiceUpdatedV3 To Request Shanghai Payload, Zero Beacon Root",
+			About: `
+			Test sending ForkchoiceUpdatedV3 to request a Shanghai payload:
+			- Payload Attributes uses Shanghai timestamp
+			- Payload Attributes' Beacon Root zero
+
+			Verify that client returns UNSUPPORTED_FORK_ERROR.
+			`,
+		},
+
+		CancunForkHeight: 2,
+
+		TestSequence: TestSequence{
+			NewPayloads{
+				FcUOnPayloadRequest: &helper.UpgradeForkchoiceUpdatedVersion{
+					ForkchoiceUpdatedCustomizer: &helper.BaseForkchoiceUpdatedCustomizer{
+						PayloadAttributesCustomizer: &helper.BasePayloadAttributesCustomizer{
+							BeaconRoot: &(common.Hash{}),
+						},
+						ExpectedError: UNSUPPORTED_FORK_ERROR,
+					},
+				},
+				ExpectationDescription: fmt.Sprintf(`
+				ForkchoiceUpdatedV3 before Cancun with beacon root must return UNSUPPORTED_FORK_ERROR (code %d)
+				`, *UNSUPPORTED_FORK_ERROR),
+			},
+		},
+	},
+
+	// ForkchoiceUpdatedV2 before cancun with beacon root
+	&CancunBaseSpec{
+		Spec: test.Spec{
+			Name: "ForkchoiceUpdatedV2 To Request Shanghai Payload, Zero Beacon Root",
+			About: `
+			Test sending ForkchoiceUpdatedV2 to request a Cancun payload:
+			- Payload Attributes uses Shanghai timestamp
+			- Payload Attributes' Beacon Root zero
+
+			Verify that client returns INVALID_PARAMS_ERROR.
+			`,
+		},
+
+		CancunForkHeight: 1,
+
+		TestSequence: TestSequence{
+			NewPayloads{
+				FcUOnPayloadRequest: &helper.DowngradeForkchoiceUpdatedVersion{
+					ForkchoiceUpdatedCustomizer: &helper.BaseForkchoiceUpdatedCustomizer{
+						PayloadAttributesCustomizer: &helper.BasePayloadAttributesCustomizer{
+							BeaconRoot: &(common.Hash{}),
+						},
+						ExpectedError: INVALID_PARAMS_ERROR,
+					},
+				},
+				ExpectationDescription: fmt.Sprintf(`
+				ForkchoiceUpdatedV2 before Cancun with beacon root field must return INVALID_PARAMS_ERROR (code %d)
+				`, *INVALID_PARAMS_ERROR),
+			},
+		},
+	},
+
+	// ForkchoiceUpdatedV2 after cancun
+	&CancunBaseSpec{
+		Spec: test.Spec{
+			Name: "ForkchoiceUpdatedV2 To Request Cancun Payload, Zero Beacon Root",
+			About: `
+			Test sending ForkchoiceUpdatedV2 to request a Cancun payload:
+			- Payload Attributes uses Cancun timestamp
+			- Payload Attributes' Beacon Root zero
```

Comment:
would like to avoid interpreting beacon root (check against zero) to return error. not essential imo but also no strong opinions here

Reply:
My intention was to test that the client responds "Invalid Params" when the beacon root is provided for FcUv2, which should not contain a beacon root, trying to follow this line on v3's spec: "provided set of parameters and their fields strictly matches the expected one".

The zero beacon root was just to represent a non-nil value, any non-nil value should (in my opinion) result in the same error.

Reply:
ahh ok thanks for clarification